llama-index-integrations/readers/llama-index-readers-zyte-serp/llama_index/readers/zyte_serp/base.py

```diff
@@ -0,0 +1,82 @@
+from typing import Literal, Optional
+from pydantic import Field
+
+from llama_index.core.readers.base import BasePydanticReader
+from llama_index.core.schema import Document
+
+
+class ZyteSerpReader(BasePydanticReader):
+    """Get google search results URLs for a search query.
+
+    Args:
+        api_key: Zyte API key.
+        extract_from: Determines the mode while extracting the .
+            It can take one of the following values: 'html', 'html-text', 'article'
+
+    Example:
+        .. code-block:: python
+
+            from llama_index.readers.zyte_serp import ZyteSerpReader
+
+            reader = ZyteSerpReader(
+               api_key="ZYTE_API_KEY",
+            )
+            docs = reader.load_data(
+                "search query",
+            )
+
+    Zyte-API reference:
+        https://www.zyte.com/zyte-api/
```

Comment:
Not sure, maybe it's better to link to the docs instead, e.g. to `https://docs.zyte.com/zyte-api/get-started.html`, as it's targeted to developers